src/Microsoft.Diagnostics.Monitoring.WebApi/Metrics/PrometheusDataModel.cs

```diff
@@ -38,17 +42,30 @@ public static string GetPrometheusNormalizedName(string metricProvider, string m
             StringBuilder builder = new StringBuilder(metricProvider.Length + metric.Length + (hasUnit ? baseUnit.Length + 1 : 0) + 1);
 
             NormalizeString(builder, metricProvider, isProvider: true);
-            builder.Append(SeperatorChar);
+            builder.Append(SeparatorChar);
             NormalizeString(builder, metric, isProvider: false);
             if (hasUnit)
             {
-                builder.Append(SeperatorChar);
+                builder.Append(SeparatorChar);
                 NormalizeString(builder, baseUnit, isProvider: false);
             }
 
             return builder.ToString();
         }
 
+        public static string GetPrometheusNormalizedLabel(string key, string value)
+        {
+            StringBuilder builder = new StringBuilder(key.Length + 2 * value.Length + 3); // Includes =,", and ", as well as extra padding for potential escape characters in the value
```

Comment:
nit: We might want to reuse the `StringBuilder` between invocations if there are many metadata pairs that are written out; this method would then call `EnsureCapacity` to make sure it has enough room for the current metadata item.

Reply:
I figure we'll want to keep this consistent with whatever we do in `GetPrometheusNormalizedName` (which is what I modeled this on), and it looks like there's already a note about potentially optimizing by using Span/stackalloc there. With that in mind, I think I'll merge this as-is, but if you're still concerned about performance, there can be another PR for investigating changing our logic in these methods.